atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/async/query/forming/QueryFormer.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.async.query.forming;
+
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+
+public interface QueryFormer {
+
+    enum SupportedQuery {
+        GET;
+
+        @Override
+        public String toString() {
+            switch (this) {
+                case GET:
+                    return "GET";
+                default:
+                    throw new RuntimeException("Method toString not defined for all enums in SupportedQuery");
```

Comment:
nit: In most usage this is a `SafeIllegalStateException`, and it would be good to include what `this` is to facilitate easier debugging.